src/ORM/Association.php

```diff
@@ -226,7 +226,7 @@ public function __construct(string $alias, array $options = [])
             }
         }
 
-        if (empty($this->_className) && strpos($alias, '.')) {
+        if (empty($this->_className)) {
             $this->_className = $alias;
```

Comment:
The `_getClassName()` function I removed the did camelize the alias but I have skipped it. Conventionally we use camel cased aliases for associations and if someone doesn't follow that convention then they should just explicitly set the `className` config.

Reply:
This check was added in #6054 but there isn't much context on why. If all the tests pass I think we should move forward, these changes make the implementation simpler.